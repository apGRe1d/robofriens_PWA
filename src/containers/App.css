/* #### Generated By: http://www.cufonfonts.com #### */

@font-face {
  font-display: swap;
  font-family: 'SEGA LOGO FONT';
  font-style: normal;
  font-weight: normal;
  src: local('SEGA LOGO FONT'), url('SEGA.woff') format('woff');
}

h1 {
  font-family: 'SEGA LOGO FONT', serif;
  font-weight: 200;
  color: #0ccac4;
}

@media screen and (max-width: 540px){
  .f1 {
    font-size: 2rem;
  }
}

@media screen and (min-width: 1650px){
  .robot-image {
    width: 300px;
    height: 300px;
  }
}

@media screen and (min-width: 1920px){
  .robot-image {
    width: 400px;
    height: 400px;
  }
}
