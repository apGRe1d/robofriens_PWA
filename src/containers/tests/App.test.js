import { shallow, mount } from 'enzyme';
import App from '../App';
import React from 'react';
import sinon from 'sinon';

describe('App.js TESTING', () => {
  let wrapper;
  beforeEach(() => {
    // монтируется перед каждым тестом
    wrapper = mount(<App />);
  });

  it('snapshot test', () => {
    expect.assertions(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('test fetching', function () {
    expect.assertions(3);
    return fetch('https://jsonplaceholder.typicode.com/users')
      .then(data => {
        return data.json();
      })
      .then(result => {
        expect(result.length).toEqual(10);
        wrapper.setState({robots: result});
        expect(wrapper.state().robots).toEqual(result);
        expect(wrapper).toMatchSnapshot();
      });
  });

  it('componentDidMount call test with UNMOUNT', function () {
    // размонтируем нашу обертку, т.к. она создается перед каждым тестом в beforeEach
    wrapper.unmount();
    // размонтирование было для того, чтобы сначала повесить шпиона на метод
    const spy = jest.spyOn(App.prototype, 'componentDidMount');
    // а уже после того, как повесили шпиона, мы можем запустить функцию маунт и создать обёртку
    // для тестирования метода componentDidMount важно использовать метод mount, а не shallow
    // т.к. метод shallow не вызывает методы жизненного цикла
    wrapper = mount(<App />);
    // тогда ожидание будет работать правильно
    expect(spy).toHaveBeenCalledTimes(1);
  });

  // it('componentDidMount call test ', function () {
    // expect.assertions(1);

    // const spy = jest.spyOn(wrapper.instance(), 'componentDidMount');
    // wrapper.instance().componentDidMount();
    // expect(spy).toHaveBeenCalledTimes(1);
    // то же самое
    // expect(wrapper.instance().componentDidMount).toHaveBeenCalledTimes(1);
  // });

  it('on search change test ', function () {
    wrapper.instance().onSearchChange({
      target: {
        value: 'asd'
      }
    })
    expect(wrapper.state().searchfield).toEqual('asd');
  });


});

