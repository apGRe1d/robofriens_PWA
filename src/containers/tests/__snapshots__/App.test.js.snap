// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js TESTING snapshot test 1`] = `
<App>
  <h1>
    Loading
  </h1>
</App>
`;

exports[`App.js TESTING test fetching 1`] = `
<App>
  <div
    className="tc"
  >
    <h1
      className="f1"
    >
      RoboFriends
    </h1>
    <SearchBox
      searchChange={[Function]}
    >
      <div
        className="pa2"
      >
        <input
          aria-label="Search Robots"
          className="pa3 ba b--green bg-lightest-blue"
          onChange={[Function]}
          placeholder="search robots"
          type="search"
        />
      </div>
    </SearchBox>
    <Scroll>
      <div
        style={
          Object {
            "border": "5px solid black",
            "height": "800px",
            "overflow": "scroll",
          }
        }
      >
        <ErrorBoundary>
          <CardList
            robots={
              Array [
                Object {
                  "address": Object {
                    "city": "Gwenborough",
                    "geo": Object {
                      "lat": "-37.3159",
                      "lng": "81.1496",
                    },
                    "street": "Kulas Light",
                    "suite": "Apt. 556",
                    "zipcode": "92998-3874",
                  },
                  "company": Object {
                    "bs": "harness real-time e-markets",
                    "catchPhrase": "Multi-layered client-server neural-net",
                    "name": "Romaguera-Crona",
                  },
                  "email": "Sincere@april.biz",
                  "id": 1,
                  "name": "Leanne Graham",
                  "phone": "1-770-736-8031 x56442",
                  "username": "Bret",
                  "website": "hildegard.org",
                },
                Object {
                  "address": Object {
                    "city": "Wisokyburgh",
                    "geo": Object {
                      "lat": "-43.9509",
                      "lng": "-34.4618",
                    },
                    "street": "Victor Plains",
                    "suite": "Suite 879",
                    "zipcode": "90566-7771",
                  },
                  "company": Object {
                    "bs": "synergize scalable supply-chains",
                    "catchPhrase": "Proactive didactic contingency",
                    "name": "Deckow-Crist",
                  },
                  "email": "Shanna@melissa.tv",
                  "id": 2,
                  "name": "Ervin Howell",
                  "phone": "010-692-6593 x09125",
                  "username": "Antonette",
                  "website": "anastasia.net",
                },
                Object {
                  "address": Object {
                    "city": "McKenziehaven",
                    "geo": Object {
                      "lat": "-68.6102",
                      "lng": "-47.0653",
                    },
                    "street": "Douglas Extension",
                    "suite": "Suite 847",
                    "zipcode": "59590-4157",
                  },
                  "company": Object {
                    "bs": "e-enable strategic applications",
                    "catchPhrase": "Face to face bifurcated interface",
                    "name": "Romaguera-Jacobson",
                  },
                  "email": "Nathan@yesenia.net",
                  "id": 3,
                  "name": "Clementine Bauch",
                  "phone": "1-463-123-4447",
                  "username": "Samantha",
                  "website": "ramiro.info",
                },
                Object {
                  "address": Object {
                    "city": "South Elvis",
                    "geo": Object {
                      "lat": "29.4572",
                      "lng": "-164.2990",
                    },
                    "street": "Hoeger Mall",
                    "suite": "Apt. 692",
                    "zipcode": "53919-4257",
                  },
                  "company": Object {
                    "bs": "transition cutting-edge web services",
                    "catchPhrase": "Multi-tiered zero tolerance productivity",
                    "name": "Robel-Corkery",
                  },
                  "email": "Julianne.OConner@kory.org",
                  "id": 4,
                  "name": "Patricia Lebsack",
                  "phone": "493-170-9623 x156",
                  "username": "Karianne",
                  "website": "kale.biz",
                },
                Object {
                  "address": Object {
                    "city": "Roscoeview",
                    "geo": Object {
                      "lat": "-31.8129",
                      "lng": "62.5342",
                    },
                    "street": "Skiles Walks",
                    "suite": "Suite 351",
                    "zipcode": "33263",
                  },
                  "company": Object {
                    "bs": "revolutionize end-to-end systems",
                    "catchPhrase": "User-centric fault-tolerant solution",
                    "name": "Keebler LLC",
                  },
                  "email": "Lucio_Hettinger@annie.ca",
                  "id": 5,
                  "name": "Chelsey Dietrich",
                  "phone": "(254)954-1289",
                  "username": "Kamren",
                  "website": "demarco.info",
                },
                Object {
                  "address": Object {
                    "city": "South Christy",
                    "geo": Object {
                      "lat": "-71.4197",
                      "lng": "71.7478",
                    },
                    "street": "Norberto Crossing",
                    "suite": "Apt. 950",
                    "zipcode": "23505-1337",
                  },
                  "company": Object {
                    "bs": "e-enable innovative applications",
                    "catchPhrase": "Synchronised bottom-line interface",
                    "name": "Considine-Lockman",
                  },
                  "email": "Karley_Dach@jasper.info",
                  "id": 6,
                  "name": "Mrs. Dennis Schulist",
                  "phone": "1-477-935-8478 x6430",
                  "username": "Leopoldo_Corkery",
                  "website": "ola.org",
                },
                Object {
                  "address": Object {
                    "city": "Howemouth",
                    "geo": Object {
                      "lat": "24.8918",
                      "lng": "21.8984",
                    },
                    "street": "Rex Trail",
                    "suite": "Suite 280",
                    "zipcode": "58804-1099",
                  },
                  "company": Object {
                    "bs": "generate enterprise e-tailers",
                    "catchPhrase": "Configurable multimedia task-force",
                    "name": "Johns Group",
                  },
                  "email": "Telly.Hoeger@billy.biz",
                  "id": 7,
                  "name": "Kurtis Weissnat",
                  "phone": "210.067.6132",
                  "username": "Elwyn.Skiles",
                  "website": "elvis.io",
                },
                Object {
                  "address": Object {
                    "city": "Aliyaview",
                    "geo": Object {
                      "lat": "-14.3990",
                      "lng": "-120.7677",
                    },
                    "street": "Ellsworth Summit",
                    "suite": "Suite 729",
                    "zipcode": "45169",
                  },
                  "company": Object {
                    "bs": "e-enable extensible e-tailers",
                    "catchPhrase": "Implemented secondary concept",
                    "name": "Abernathy Group",
                  },
                  "email": "Sherwood@rosamond.me",
                  "id": 8,
                  "name": "Nicholas Runolfsdottir V",
                  "phone": "586.493.6943 x140",
                  "username": "Maxime_Nienow",
                  "website": "jacynthe.com",
                },
                Object {
                  "address": Object {
                    "city": "Bartholomebury",
                    "geo": Object {
                      "lat": "24.6463",
                      "lng": "-168.8889",
                    },
                    "street": "Dayna Park",
                    "suite": "Suite 449",
                    "zipcode": "76495-3109",
                  },
                  "company": Object {
                    "bs": "aggregate real-time technologies",
                    "catchPhrase": "Switchable contextually-based project",
                    "name": "Yost and Sons",
                  },
                  "email": "Chaim_McDermott@dana.io",
                  "id": 9,
                  "name": "Glenna Reichert",
                  "phone": "(775)976-6794 x41206",
                  "username": "Delphine",
                  "website": "conrad.com",
                },
                Object {
                  "address": Object {
                    "city": "Lebsackbury",
                    "geo": Object {
                      "lat": "-38.2386",
                      "lng": "57.2232",
                    },
                    "street": "Kattie Turnpike",
                    "suite": "Suite 198",
                    "zipcode": "31428-2261",
                  },
                  "company": Object {
                    "bs": "target end-to-end models",
                    "catchPhrase": "Centralized empowering task-force",
                    "name": "Hoeger LLC",
                  },
                  "email": "Rey.Padberg@karina.biz",
                  "id": 10,
                  "name": "Clementina DuBuque",
                  "phone": "024-648-3804",
                  "username": "Moriah.Stanton",
                  "website": "ambrose.net",
                },
              ]
            }
          >
            <div>
              <Card
                email="Sincere@april.biz"
                id={1}
                key="0"
                name="Leanne Graham"
              >
                <div
                  className="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
                >
                  <picture>
                    <source
                      height="200px"
                      srcSet="
            https://robohash.org/1.avif 200w,
            https://robohash.org/1.avif 300w,
            https://robohash.org/1.avif 400w,
          "
                      type="image/avif"
                      width="200px"
                    />
                    <img
                      alt="robots"
                      className="robot-image"
                    />
                  </picture>
                  <div>
                    <h2>
                      Leanne Graham
                    </h2>
                    <p>
                      Sincere@april.biz
                    </p>
                  </div>
                </div>
              </Card>
              <Card
                email="Shanna@melissa.tv"
                id={2}
                key="1"
                name="Ervin Howell"
              >
                <div
                  className="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
                >
                  <picture>
                    <source
                      height="200px"
                      srcSet="
            https://robohash.org/2.avif 200w,
            https://robohash.org/2.avif 300w,
            https://robohash.org/2.avif 400w,
          "
                      type="image/avif"
                      width="200px"
                    />
                    <img
                      alt="robots"
                      className="robot-image"
                    />
                  </picture>
                  <div>
                    <h2>
                      Ervin Howell
                    </h2>
                    <p>
                      Shanna@melissa.tv
                    </p>
                  </div>
                </div>
              </Card>
              <Card
                email="Nathan@yesenia.net"
                id={3}
                key="2"
                name="Clementine Bauch"
              >
                <div
                  className="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
                >
                  <picture>
                    <source
                      height="200px"
                      srcSet="
            https://robohash.org/3.avif 200w,
            https://robohash.org/3.avif 300w,
            https://robohash.org/3.avif 400w,
          "
                      type="image/avif"
                      width="200px"
                    />
                    <img
                      alt="robots"
                      className="robot-image"
                    />
                  </picture>
                  <div>
                    <h2>
                      Clementine Bauch
                    </h2>
                    <p>
                      Nathan@yesenia.net
                    </p>
                  </div>
                </div>
              </Card>
              <Card
                email="Julianne.OConner@kory.org"
                id={4}
                key="3"
                name="Patricia Lebsack"
              >
                <div
                  className="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
                >
                  <picture>
                    <source
                      height="200px"
                      srcSet="
            https://robohash.org/4.avif 200w,
            https://robohash.org/4.avif 300w,
            https://robohash.org/4.avif 400w,
          "
                      type="image/avif"
                      width="200px"
                    />
                    <img
                      alt="robots"
                      className="robot-image"
                    />
                  </picture>
                  <div>
                    <h2>
                      Patricia Lebsack
                    </h2>
                    <p>
                      Julianne.OConner@kory.org
                    </p>
                  </div>
                </div>
              </Card>
              <Card
                email="Lucio_Hettinger@annie.ca"
                id={5}
                key="4"
                name="Chelsey Dietrich"
              >
                <div
                  className="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
                >
                  <picture>
                    <source
                      height="200px"
                      srcSet="
            https://robohash.org/5.avif 200w,
            https://robohash.org/5.avif 300w,
            https://robohash.org/5.avif 400w,
          "
                      type="image/avif"
                      width="200px"
                    />
                    <img
                      alt="robots"
                      className="robot-image"
                    />
                  </picture>
                  <div>
                    <h2>
                      Chelsey Dietrich
                    </h2>
                    <p>
                      Lucio_Hettinger@annie.ca
                    </p>
                  </div>
                </div>
              </Card>
              <Card
                email="Karley_Dach@jasper.info"
                id={6}
                key="5"
                name="Mrs. Dennis Schulist"
              >
                <div
                  className="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
                >
                  <picture>
                    <source
                      height="200px"
                      srcSet="
            https://robohash.org/6.avif 200w,
            https://robohash.org/6.avif 300w,
            https://robohash.org/6.avif 400w,
          "
                      type="image/avif"
                      width="200px"
                    />
                    <img
                      alt="robots"
                      className="robot-image"
                    />
                  </picture>
                  <div>
                    <h2>
                      Mrs. Dennis Schulist
                    </h2>
                    <p>
                      Karley_Dach@jasper.info
                    </p>
                  </div>
                </div>
              </Card>
              <Card
                email="Telly.Hoeger@billy.biz"
                id={7}
                key="6"
                name="Kurtis Weissnat"
              >
                <div
                  className="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
                >
                  <picture>
                    <source
                      height="200px"
                      srcSet="
            https://robohash.org/7.avif 200w,
            https://robohash.org/7.avif 300w,
            https://robohash.org/7.avif 400w,
          "
                      type="image/avif"
                      width="200px"
                    />
                    <img
                      alt="robots"
                      className="robot-image"
                    />
                  </picture>
                  <div>
                    <h2>
                      Kurtis Weissnat
                    </h2>
                    <p>
                      Telly.Hoeger@billy.biz
                    </p>
                  </div>
                </div>
              </Card>
              <Card
                email="Sherwood@rosamond.me"
                id={8}
                key="7"
                name="Nicholas Runolfsdottir V"
              >
                <div
                  className="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
                >
                  <picture>
                    <source
                      height="200px"
                      srcSet="
            https://robohash.org/8.avif 200w,
            https://robohash.org/8.avif 300w,
            https://robohash.org/8.avif 400w,
          "
                      type="image/avif"
                      width="200px"
                    />
                    <img
                      alt="robots"
                      className="robot-image"
                    />
                  </picture>
                  <div>
                    <h2>
                      Nicholas Runolfsdottir V
                    </h2>
                    <p>
                      Sherwood@rosamond.me
                    </p>
                  </div>
                </div>
              </Card>
              <Card
                email="Chaim_McDermott@dana.io"
                id={9}
                key="8"
                name="Glenna Reichert"
              >
                <div
                  className="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
                >
                  <picture>
                    <source
                      height="200px"
                      srcSet="
            https://robohash.org/9.avif 200w,
            https://robohash.org/9.avif 300w,
            https://robohash.org/9.avif 400w,
          "
                      type="image/avif"
                      width="200px"
                    />
                    <img
                      alt="robots"
                      className="robot-image"
                    />
                  </picture>
                  <div>
                    <h2>
                      Glenna Reichert
                    </h2>
                    <p>
                      Chaim_McDermott@dana.io
                    </p>
                  </div>
                </div>
              </Card>
              <Card
                email="Rey.Padberg@karina.biz"
                id={10}
                key="9"
                name="Clementina DuBuque"
              >
                <div
                  className="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
                >
                  <picture>
                    <source
                      height="200px"
                      srcSet="
            https://robohash.org/10.avif 200w,
            https://robohash.org/10.avif 300w,
            https://robohash.org/10.avif 400w,
          "
                      type="image/avif"
                      width="200px"
                    />
                    <img
                      alt="robots"
                      className="robot-image"
                    />
                  </picture>
                  <div>
                    <h2>
                      Clementina DuBuque
                    </h2>
                    <p>
                      Rey.Padberg@karina.biz
                    </p>
                  </div>
                </div>
              </Card>
            </div>
          </CardList>
        </ErrorBoundary>
      </div>
    </Scroll>
  </div>
</App>
`;
